specification/cdn/resource-manager/Microsoft.Cdn/preview/2019-06-15-preview/cdn.json

```diff
@@ -0,0 +1,4353 @@
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Cdn/checkResourceUsage": {
+      "post": {
+        "tags": [
+          "CheckResourceUsage"
+        ],
+        "description": "Check the quota and actual usage of the CDN profiles under the given subscription.",
+        "operationId": "ResourceUsage_List",
+        "x-ms-examples": {
+          "ResourceUsage_List": {
+            "$ref": "./examples/ResourceUsage_List.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ResourceUsageListResult"
+            }
+          },
+          "default": {
+            "description": "CDN error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Cdn/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "description": "Lists all of the available CDN REST API operations.",
+        "operationId": "Operations_List",
+        "x-ms-examples": {
+          "Operations_List": {
+            "$ref": "./examples/Operations_List.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/OperationsListResult"
+            }
+          },
+          "default": {
+            "description": "CDN error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Cdn/edgenodes": {
+      "get": {
+        "tags": [
+          "Edgenodes"
+        ],
+        "description": "Edgenodes are the global Point of Presence (POP) locations used to deliver CDN content to end users.",
+        "operationId": "EdgeNodes_List",
+        "x-ms-examples": {
+          "EdgeNodes_List": {
+            "$ref": "./examples/EdgeNodes_List.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200"
```

Comment:
This should be a duration, not a string.

Reply:
@devigned I am not sure what you meant on here, can you please clarify if posible.
Also, this is completely unrelated to me, all I have done is copied the cdn.json from the stable folder and added one line to the swagger which was needed to add a link from cdn endpoint to the cdnWebApplicationFirewallPolicies. You can find that line by searching for "cdnWebApplicationFirewallPolicies"

Reply:
This was meant for cache duration: https://github.com/Azure/azure-rest-api-specs/pull/5885/files/7f720a74e69ddb58977ad020865a53d5f24a06cf#diff-6d53702dc9b0abd52dedefbde8482955R3372. I'm not sure why the line was marked at 1948.

Reply:
The cacheDuration should be https://en.wikipedia.org/wiki/ISO_8601#Durations

Reply:
@devigned ok got it. In this case, this is what is in the stable cdn version 
https://github.com/sushantsingh/azure-rest-api-specs/blob/master/specification/cdn/resource-manager/Microsoft.Cdn/stable/2019-04-15/cdn.json

My only change to cdn.json (other than copying it to preview folder) is
        "webApplicationFirewallPolicyLink": {
          "description": "Defines the Web Application Firewall policy for the endpoint (if applicable)",
          "type": "object",
          "properties": {
            "id": {
              "type": "string",
              "description": "Resource ID."
            }
          }
        }

Reply:
I understand, but that doesn't change that it should be a ISO 8601 duration.

Reply:
I meant to say above that it is part of the existing cdn.json swagger and is not introduced by me. The reason it shows up in the diff is because the one change I did make to the cdn swagger needs to be a preview change as well, so I had to copy the entire thing into the preview folder.

Reply:
Will start a email thread with David Justice and the owner of cdn.json